pkl-doc/src/main/kotlin/org/pkl/doc/PageGenerator.kt

```diff
@@ -525,15 +525,30 @@ internal abstract class PageGenerator<out S>(
         var first = true
         for (dep in dependencies) {
           if (first) first = false else +", "
-          a {
-            href =
-              dep.documentation?.toString()
-                ?: pageScope.relativeSiteUrl
-                  .resolve("${dep.name}/${dep.version}/index.html")
-                  .toString()
-            +dep.name
-            +":"
-            +dep.version
+          val text = "${dep.name}:${dep.version}"
+          if (dep.documentation != null) {
+            a {
+              href = dep.documentation.toString()
+              +text
+            }
+          } else {
+            val localSitePackage =
+              pageScope.siteScope?.packageScopes?.values?.find {
+                it.docPackageInfo.name == dep.name
+              }
+            if (localSitePackage != null) {
+              a {
+                href =
+                  pageScope.relativeSiteUrl
+                    .resolve(
+                      "${localSitePackage.docPackageInfo.name.pathEncoded}/current/index.html"
+                    )
```

Comment:
This will only show a local link if the package is known to Pkldoc, and it will link to "current" rather than the depended upon version. If the package is not known, _and_ we don't know what its documentation URL is, this will just display plain text.

Note: I considered making this link directly to the version, e.g. `"${localSitePackage.docPackageInfo.name.pathEncoded}/${localSitePackage.docPackageInfo.version.pathEncoded}/index.htm`

However, we don't know if that version exists in the doc site. The page generator doesn't know about all the versions of each package, and we don't build this index until _after_ the static HTML has been built.